metricflow/naming/linkable_spec_name.py

```diff
@@ -51,6 +99,21 @@ def from_name(qualified_name: str) -> StructuredLinkableSpecName:
         else:
             return StructuredLinkableSpecName(tuple(name_parts[:-1]), name_parts[-1])
 
+    @staticmethod
+    def from_name_v2(modified_name: str) -> StructuredLinkableSpecName:
+        """Construct from a name with newer syntax. e.g. listing__ds.granularity(month)."""
+
+        structured_modified_spec = StructuredModifiedSpecName.from_name(modified_name)
+        linked_name = structured_modified_spec.qualified_name.split(DUNDER)
```

Comment:
Why do we still need to `split` ?

Reply:
there are other identifier links that still use the dunder, which we don't change, like user__country__created_on.granularity(week) for example.

Reply:
gotcha! thanks for the clarification!